Talbs: Define Base Elements

talbs opened the pull request:
This work is an extension to https://github.com/edx/ux-pattern-library/pull/6. It offers an alternative base elements set up.

clrux commented on _src/static/sass/utilities/_helpers.scss:
```diff
@@ -0,0 +1,139 @@
+// #UTILITIES
+// ----------------------------
+// visual link
+%fake-link {
```
What would be the use case for this? Makes me kinda nervous...

talbs replied:
The main use case is if we need to style something that doesn't natively give us the cursor property like this - Its more to cover our butts for never interactions (think sliders or other elements). Happy to remove it if it really gives you the creeps - right now its not used and this will never make it to our compiled CSS as a Sass extend.

clrux replied:
I'd prefer to remove it - it's only a single style, after all - and define `cursor: pointer` on an as-needed basis. You cool with that?

talbs replied:
Sure, will do.

clrux commented on _src/static/sass/utilities/_helpers.scss:
```diff
@@ -0,0 +1,139 @@
+
+// accessibility
+%a11y-hide {
+  border: 0;
```
NIT: Spacing

clrux commented on _src/static/sass/utilities/_helpers.scss:
```diff
@@ -0,0 +1,139 @@
+// ----------------------------
+// reset of lists
+%no-list {
+  list-style: none;
```
NIT: Spacing

clrux commented on _src/static/sass/utilities/_helpers.scss:
```diff
@@ -0,0 +1,139 @@
+// #UTILITIES
+// ----------------------------
+// reset of lists
+%no-list {
```
Is `no-list` accurate? This would go _on_ a list, no? I feel like this lacks semantics.

talbs replied:
Good feedback - I'll roll the properties here into ``%reset-list`` extend.

clrux commented on _src/static/sass/utilities/_helpers.scss:
```diff
@@ -0,0 +1,139 @@
+
+// reset of links
+%link {
+    @include transition(all 0.15s ease-in-out);
```
Should we use one of the timing variables here?

talbs replied:
Yep, totes!

clrux commented on _src/static/sass/utilities/_variables.scss:
```diff
@@ -3,156 +3,316 @@
+// colors: palettes
+$palettes: (
+    primary:        (
+        base:       $brand-primary,
```
For each of these colors, how many variations will we really need?

talbs replied:
Good question - This is creating monochromatic scales of values per color (the actual color base is  defined/abstracted in good old Sass variables beforehand, for easier configuration and theming). I'm fine with pairing down the computed scale once we start using them, but this gives us different shades of each base color (primary, secondary, accent, black-white scale, and system colors like red, green, yellow) which seems barebones enough.

clrux commented on _src/static/sass/utilities/_variables.scss:
```diff
@@ -3,156 +3,316 @@
+// scale roots
+$font-scale-small: 14;
+$font-scale-medium: 16;
+$font-scale-large: 16;
```
Large is the same as medium; do we need them both?

talbs replied:
yes, these are roots that will be used for different breakpoint/response font bases - the scale will flex depending on this initial value. So, medium will be used for medium-based viewport sizes and large will be used for large.

clrux commented on _src/static/sass/utilities/_variables.scss:
```diff
@@ -3,156 +3,316 @@
+// #TIMING
+// ----------------------------
+$timing: (
+    base:       0.75s,
```
Do we need this many variations? Maybe for visual consistency we limit it to three: slow, normal, fast?

talbs replied:
Good question - I'd prefer us to use them some and see how many we end up referencing. I know in the past, the slower ones have been used for larger UI animations/transitions while the very fast ones are used for basic transitions. Let's revisit as we get further in PL work.

clrux replied:
What about slimming it down now and seeing if we need more and if so, add them? I fear that if we left them, we'd forget to go back and slim down. Adding as we need ensures that we never had too much, and means we won't have to search/find through the entire app to see if this or that is being used.

talbs replied:
Sure, will do.

talbs commented:
@clrux, I've addressed your comments (thanks for the thoughts/feedback). For the items that were straightforward, I've made changes. For the larger questions, I've given my thoughts. Let me know if you have any more recommendations or hesitations on those.

talbs commented:
@clrux, I've fixed those two additional requests you had. Mind giving this a look/thumbs for merging?

talbs commented:
Got the verbal thumbs, so merging. Thanks!
